src/modules/hip/kernel/spatter.hpp

```diff
@@ -0,0 +1,322 @@
+#include <hip/hip_runtime.h>
+#include "hip/rpp_hip_common.hpp"
+#include "func_specific/spatter_mask.hpp"
+#include <random>
+
+__device__ void spatter_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *mask_f8, d_float8 *maskInv_f8, float4 *pix_f4)
+{
+    dst_f8->x = (src_f8->x * maskInv_f8->x) + (*pix_f4 * mask_f8->x);
+    dst_f8->y = (src_f8->y * maskInv_f8->y) + (*pix_f4 * mask_f8->y);
+}
+
+__device__ void spatter_hip_compute(float *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *mask_f8, d_float8 *maskInv_f8, float4 *pix_f4)
+{
+    float4 pixNorm_f4 = *pix_f4 * (float4)0.0039216;
+    dst_f8->x = (src_f8->x * maskInv_f8->x) + (pixNorm_f4 * mask_f8->x);
+    dst_f8->y = (src_f8->y * maskInv_f8->y) + (pixNorm_f4 * mask_f8->y);
+}
+
+__device__ void spatter_hip_compute(schar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *mask_f8, d_float8 *maskInv_f8, float4 *pix_f4)
+{
+    dst_f8->x = ((src_f8->x + (float4)128) * maskInv_f8->x) + (*pix_f4 * mask_f8->x) - (float4)128;
+    dst_f8->y = ((src_f8->y + (float4)128) * maskInv_f8->y) + (*pix_f4 * mask_f8->y) - (float4)128;
+}
+
+__device__ void spatter_hip_compute(half *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *mask_f8, d_float8 *maskInv_f8, float4 *pix_f4)
+{
+    float4 pixNorm_f4 = *pix_f4 * (float4)0.0039216;
```

Comment:
what's this 0.0039216 magic number? can you add a comment for every const number used in these kernels?

Reply:
Done. #define is now used to initialize a common 1/255